test/service_config_deprecated_test.go

```diff
@@ -0,0 +1,456 @@
+/*
```

Comment:
Nit: the file name feels weird to me. I see a comment on line 34 as such: "
// The following functions with function name ending with TD indicates that they
// should be deleted after old service config API is deprecated and deleted." Thus, we're not testing deprecated functionality, but functionality which we plan to deprecate?

Reply:
Yes, this API is already deprecated. See: https://github.com/grpc/grpc-go/blob/master/dialoptions.go#L237-L248.